Frontends/CMakeLists.txt

```diff
@@ -19,6 +19,20 @@ if(BUILD_CSOUND_COMMAND)
     target_link_libraries(csound-bin PRIVATE Threads::Threads)
   endif()
 
+  if(HAVE_UNISTD_H)
+    target_compile_definitions(csound-bin PRIVATE "HAVE_UNISTD_H")
+  endif()
+
+  if(WIN32)
+    target_compile_definitions(csound-bin PRIVATE "WIN32")
+  endif()
+  if(LINUX)
+    target_compile_definitions(csound-bin PRIVATE "LINUX")
+  endif()
+  if(MACOSX)
+    target_compile_definitions(csound-bin PRIVATE "MACOSX")
+  endif()
```

Comment:
I feel like there is a more natural way of doing this surely

Reply:
Yeah I think so too. Compilers should already set macros to identify the OS and we should probably use those instead.

Reply:
But that can wait for another PR?